### ISSUE TITLE
onDelete question

### ISSUE THREAD

**Issue by radoslavpetranov** [sequelize/sequelize#3031]
Hello,

I'm having an issue with the onDelete param and I'm not sure if I'm doing something wrong or if this is a bug (most likely the former).

I have this hasMany definition

```
this.dbObjects.ModelA.hasMany(this.dbObjects.ModelB, {as:'entries', foreignKey: 'model_a_id', foreignKeyConstraint: true, onDelete : 'cascade'});
```

and then I have some code that creates an array of ModelA ids that need to be removed so I run the following command

```
return _this.dbObjects.ModelA.destroy({where : {id: idsToDeleteArray}, transaction : t})
```

which in turn generates this SQL

```
'DELETE FROM `model_a` WHERE `id` IN (524,560)'
```

which then fails because of a foreign key constraint saying that 

```
{ [Error: ER_ROW_IS_REFERENCED_: Cannot delete or update a parent row: a foreign key constraint fails (`my_db`.`model_b`, CONSTRAINT `model_b_ibfk_3` FOREIGN KEY (`model_a_id`) REFERENCES `model_a`)]
  code: 'ER_ROW_IS_REFERENCED_',
  errno: 1451,
  sqlState: '23000',
  index: 0,
  sql: 'DELETE FROM `model_a` WHERE `id` IN (524,560)' }
```

Am I wrong in thinking that the onDelete: cascade should have taken care of this?

Thanks and cheers!

**Comment by mickhansen** [comment]
It helps everyone if you put in the effort and format your issue correctly with markdown :)

**Comment by mickhansen** [comment]
The cascade should indeed be taking care of that, what does the `sync()` log look like?

**Comment by radoslavpetranov** [comment]
O wow thanks for that - I had no idea how to do that but googling around for github markdown got that figured out for me :) I had tried using <code></code> before but since that didn't work I just kept on copy/pasting stuff.

The log is fine and shows exactly what it should show - I just didn't realize this was implemented like that (I thought there was like an internal mechanism that checks if child model instances exist and drops them before doing a delete on the parent object). All I have to do is to modify our existing foreign key definitions to support onDelete cascade in mysql and we're good to go.

Cheers and thanks
